RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalUncalibRecHitPhase2WeightsProducerPortable.cc

```diff
@@ -0,0 +1,141 @@
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHitHostCollection.h"
+#include "DataFormats/EcalRecHit/interface/alpaka/EcalUncalibratedRecHitDeviceCollection.h"
+#include "DataFormats/Portable/interface/PortableObject.h"
+
+#include "EcalUncalibRecHitPhase2WeightsAlgoPortable.h"
+#include "EcalUncalibRecHitPhase2WeightsStruct.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class EcalUncalibRecHitPhase2WeightsProducerPortable : public global::EDProducer<> {
+  public:
+    explicit EcalUncalibRecHitPhase2WeightsProducerPortable(edm::ParameterSet const &ps);
+    ~EcalUncalibRecHitPhase2WeightsProducerPortable() override = default;
+    static void fillDescriptions(edm::ConfigurationDescriptions &);
+
+    void produce(edm::StreamID sid, device::Event &, device::EventSetup const &) const override;
+
+  private:
+    using InputProduct = EcalDigiPhase2DeviceCollection;
+    const device::EDGetToken<InputProduct> digisToken_;  //both tokens stored on the device
+    using OutputProduct = EcalUncalibratedRecHitDeviceCollection;
+    const device::EDPutToken<OutputProduct> uncalibratedRecHitsToken_;
+
+    // class data member
+    cms::alpakatools::MoveToDeviceCache<Device, PortableHostObject<EcalUncalibRecHitPhase2Weights>> weightsCache_;
+  };
+
+  // constructor with initialisation of elements
+  EcalUncalibRecHitPhase2WeightsProducerPortable::EcalUncalibRecHitPhase2WeightsProducerPortable(
+      const edm::ParameterSet &ps)
+      : EDProducer(ps),
+        digisToken_{consumes(ps.getParameter<edm::InputTag>("digisLabelEB"))},
+        uncalibratedRecHitsToken_{produces(ps.getParameter<std::string>("uncalibratedRecHitsLabelEB"))},
+        weightsCache_(PortableHostObject<EcalUncalibRecHitPhase2Weights>(
+            cms::alpakatools::host(), [](const edm::ParameterSet &ps) {
+              EcalUncalibRecHitPhase2Weights weights;
+              const auto amp_weights = ps.getParameter<std::vector<double>>("weights");
+              const auto timeWeights = ps.getParameter<std::vector<double>>("timeWeights");
+              for (unsigned int i = 0; i < ecalPh2::sampleSize; ++i) {
+                if (i < amp_weights.size()) {
+                  weights.weights[i] = static_cast<float>(amp_weights[i]);
+                } else {
+                  weights.weights[i] = 0;
+                }
+                if (i < timeWeights.size()) {
+                  weights.timeWeights[i] = static_cast<float>(timeWeights[i]);
+                } else {
+                  weights.timeWeights[i] = 0;
+                }
+              }
+              return weights;
+            }(ps))) {}
+
+  void EcalUncalibRecHitPhase2WeightsProducerPortable::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<std::string>("uncalibratedRecHitsLabelEB", "EcalUncalibRecHitsEB");
+    //The below weights values should be kept up to date with those on the CPU version of this module
```

Comment:
Could you add where the CPU version of the weights are ?
For example,
```suggestion
    //The weights values below should be kept up to date with those on the CPU version of this module
    //in RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsProducer.cc
```